components-core/src/main/java/com/adyen/checkout/components/core/PaymentMethod.kt

```diff
@@ -20,18 +21,25 @@ import kotlinx.parcelize.Parcelize
 import org.json.JSONException
 import org.json.JSONObject
 
+@OptIn(IgnoredCustomizedField::class)
 @Parcelize
 data class PaymentMethod(
-    var type: String? = null,
-    var name: String? = null,
-    var brands: List<String>? = null,
-    var brand: String? = null,
-    var fundingSource: String? = null,
-    var issuers: List<Issuer>? = null,
-    var configuration: Configuration? = null,
-    var details: List<InputDetail>? = null,
+    val type: String? = null,
+    @IgnoredCustomizedField
+    val name: String? = null,
+    val brands: List<String>? = null,
+    val brand: String? = null,
+    val fundingSource: String? = null,
+    val issuers: List<Issuer>? = null,
+    val configuration: Configuration? = null,
+    val details: List<InputDetail>? = null,
+    // This property is used to allow setting a customizable fields for the payment method
+    var customDisplayInformation: PaymentMethodCustomDisplayInformation? = null
```

Comment:
What about we just use the `name` field for all of this? We could override the name in `addPaymentMethodCustomDisplayInformation` and still have the flexibility of `PaymentMethodCustomDisplayInformation`, but avoid adding any logic to this class

Reply:
Why I chose to go with this approach is because `name` field is what we get from backend and when serializing, we use that field.
I think it is best to keep `name` field unchanged, for some custom use-cases, perhaps merchants want to give a custom display information by name (through creating a custom predicate). Or they have some other logic around the `name` field?

Reply:
I see and that makes sense, but the `name` field is only ment for display purposes. `name` is localised in the backend, so it would be difficult to built logic around it (`type` should be used for that). And `name` is never send back to the backend, so it shouldn't impact anything. Let's maybe discuss with the team as well